packages/core/strapi/src/cli/commands/generate-schema.ts

```diff
@@ -0,0 +1,210 @@
+    } = contentType;
+
+    const formattedSingularName = singularName
+      .split('-')
+      .map((part) => part.charAt(0).toUpperCase() + part.slice(1))
+      .join('');
+
+    const typeName = `Api${formattedSingularName}${formattedSingularName}`;
+    uidToGeneratedTypeName.set(uid, typeName);
+
+    return typeName;
+  });
+
+  // Generate a single import statement for generated types
+  const combinedImportStatement = `import { ${allTypeNames.join(', ')} } from './generated/contentTypes';`;
+
+  // Helper function to generate TypeScript interface definitions
+  const generateInterfaceDefinition = (
+    uid: string,
+    contentType: Struct.CollectionTypeSchema | Struct.SingleTypeSchema,
+    operations: string[],
+    uidToGeneratedTypeName: Map<string, string>
+  ) => {
+    const {
+      info: { singularName, pluralName },
+      kind,
+    } = contentType;
+
+    const typeName = uidToGeneratedTypeName.get(uid);
+    const operationsDefinition = operations
+      .map((op) => `${op}(): Promise<${typeName}>;`)
+      .join('\n');
+
+    const interfaceNameForCrudTypes = (
+      kind === 'collectionType' ? pluralName : singularName
+    ).replace(/-/g, '');
+
+    return `interface ${interfaceNameForCrudTypes} {\n${operationsDefinition}\n}`;
+  };
+
+  // Generate TypeScript definitions for CRUD operations
+  const collectionTypeDefinitions = apiContentTypes
+    .filter(([, contentType]) => contentType.kind === 'collectionType')
+    .map(([uid, contentType]) =>
+      generateInterfaceDefinition(
+        uid,
+        contentType,
+        crudOperationNames.collectionType,
+        uidToGeneratedTypeName
+      )
+    )
+    .join('\n\n');
+
+  const singleTypeDefinitions = apiContentTypes
+    .filter(([, contentType]) => contentType.kind === 'singleType')
+    .map(([uid, contentType]) =>
+      generateInterfaceDefinition(
+        uid,
+        contentType,
+        crudOperationNames.singleType,
+        uidToGeneratedTypeName
+      )
+    )
+    .join('\n\n');
+
+  // Combine the sections into 'CollectionTypes' and 'SingleTypes' interfaces
+  const schemaDefinitions = `
+    ${collectionTypeDefinitions}\n\n
+    ${singleTypeDefinitions}\n\n
+    export interface CollectionTypes {\n${apiContentTypes
+      .filter(([, contentType]) => contentType.kind === 'collectionType')
+      .map(
+        ([, contentType]) =>
+          `${contentType.info.pluralName.replace(/-/g, '')}: ${contentType.info.pluralName.replace(/-/g, '')}`
+      )
+      .join(';\n')}\n}\n\n
+    export interface SingleTypes {\n${apiContentTypes
+      .filter(([, contentType]) => contentType.kind === 'singleType')
+      .map(
+        ([, contentType]) =>
+          `${contentType.info.singularName.replace(/-/g, '')}: ${contentType.info.singularName.replace(/-/g, '')}`
+      )
+      .join(';\n')}\n}
+  `;
+
+  // Determine output directory
+  const outputPath = `${outputDir}/schema.ts`;
+
+  const formattedOutput = await prettier.format(
```

Comment:
Can we include a "version" that is not necessarily the strapi version (in fact, that could be a security risk since this file is public) but a file-format version, just in case we need to make structural changes without it breaking?  For now we can just fix it at "0.0.1" and hopefully never change it.